Connectors/Rhino/Speckle.Connectors.GrasshopperShared/Parameters/Wrappers/SpeckleDataObjectWrapperGoo.cs

```diff
@@ -63,4 +137,60 @@ public override bool CastTo<T>(ref T target) =>
 
   // TODO: CNX-2094
   public BoundingBox ClippingBox { get; }
+
+  /// <summary>
+  /// Creates a single-element DataObject from <see cref="SpeckleGeometryWrapper"/> (one geometry → one display value).
+  /// </summary>
+  private bool CastFromSpeckleGeometryWrapper(SpeckleGeometryWrapper geometryWrapper)
+  {
+    try
+    {
+      // create DataObject with single displayValue
+      DataObject dataObject =
+        new()
+        {
+          name = geometryWrapper.Name,
+          displayValue = [geometryWrapper.Base],
+          properties = geometryWrapper.Properties.Unwrap(),
+          applicationId = geometryWrapper.ApplicationId
+        };
+
+      // create wrapper - Name, ApplicationId and Properties kept in sync with wrapped DataObject through getters/setters
+      // geometry will inherit DataObject properties through the syncing (hopefully)
+      Value = new SpeckleDataObjectWrapper { Base = dataObject, Geometries = [geometryWrapper] };
+
+      return true;
+    }
+    catch (Exception ex) when (!ex.IsFatal())
+    {
+      return false;
+    }
+  }
+
+  private bool CastFromIghGeometricGoo(IGH_GeometricGoo geometricGoo)
+  {
+    try
+    {
+      // reuse existing logic from SpeckleGeometryWrapperGoo
+      GeometryBase gb = geometricGoo.ToGeometryBase();
+      Base converted = SpeckleConversionContext.ConvertToSpeckle(gb);
+      string appId = Guid.NewGuid().ToString();
+
+      // create geometry wrapper first
+      var geometryWrapper = new SpeckleGeometryWrapper
+      {
+        GeometryBase = gb,
+        Base = converted,
+        ApplicationId = appId,
+        Name = $"Converted {geometricGoo.TypeName}"
+      };
+
+      // delegate to existing method
+      return CastFromSpeckleGeometryWrapper(geometryWrapper);
```

Comment:
This method could be simplified to:
```cs
SpeckleGeometryWrapperGoo geoGoo = new();
if (geoGoo.CastFrom(geometricGoo))
{
return CastFrom(geoGoo);
}
```

Reply:
niice. delegated to existing tested logic 👍🏼